checker/raw_result.go

```diff
@@ -226,6 +227,40 @@ type SecurityPolicyFile struct {
 	File File
 }
 
+// SASTData contains the raw results
+// for the SAST check.
+type SASTData struct {
+	Workflows    []SASTWorkflow
+	Commits      []SASTCommit
+	NumWorkflows int
+}
+
+type SASTCommit struct {
+	CommittedDate          time.Time
+	Message                string
+	SHA                    string
+	CheckRuns              []clients.CheckRun
+	AssociatedMergeRequest clients.PullRequest
+	Committer              clients.User
+	Compliant              bool
+}
+
+// SASTWorkflowType represents a type of SAST workflow.
+type SASTWorkflowType string
+
+const (
+	// CodeQLWorkflow represents a workflow that runs CodeQL.
+	CodeQLWorkflow SASTWorkflowType = "CodeQL"
+	// SonarWorkflow represents a workflow that runs Sonar.
+	SonarWorkflow SASTWorkflowType = "Sonar"
+)
```

Comment:
> Are these workflows distinct enough that they require different types?

They have different types because they are consumed by different probes.

Reply:
Why can't both probes consume a single type?

Reply:
The check distinguishes between CodeQL and Sonatype workflows and consider both in the evaluation. Essentially, the types are there to make it easy and simple to make a decision in the probes: For example, because the types are there, the probes can check for a workflow of a given type. How would we detect whether the project uses CodeQL or Sonatype if we have a single type?

Could you elaborate a bit what you are thinking of in terms of having a single (or no) type? I may be caught up in the current implementation, but I don't fully follow.

Reply:
SGTM. I think I was misguided on the comment, and I never made it far enough in the PR to correct my misunderstanding.